openlr/candidate_points_getter.cpp

```diff
@@ -0,0 +1,97 @@
+#include "openlr/candidate_points_getter.hpp"
+#include "openlr/helpers.hpp"
+#include "openlr/timer.hpp"
+
+#include "routing/road_graph.hpp"
+
+#include "storage/country_info_getter.hpp"
+
+using namespace routing;
+
+namespace
+{
+MwmSet::MwmId GetMwmIdByPoint(m2::PointD const & p, Index const & index)
+{
+  // TODO(mgsergio): Should I cache it later?
+  auto const infoGetter =
+      storage::CountryInfoReader::CreateCountryInfoReaderTwoComponentMwms(GetPlatform());
+  auto const & countryId = infoGetter->GetRegionCountryId(p);
+  ASSERT(!countryId.empty(), ("CountryId is \"\" for point", MercatorBounds::ToLatLon(p)));
+  auto const & countryFile = platform::CountryFile(countryId);
+  return index.GetMwmIdByCountryFile(countryFile);
+}
+}  // namespace
+
+namespace openlr
+{
+void CandidatePointsGetter::GetJunctionPointCandidates(m2::PointD const & p,
+                                                       vector<m2::PointD> & candidates)
+{
+  Timer t(m_stat.m_pointCadtidatesTime);
+
+  size_t const kRectSideMeters = 110;
+
+  auto const mwmId = GetMwmIdByPoint(p, m_index);
+
+  auto const rect = MercatorBounds::RectByCenterXYAndSizeInMeters(p, kRectSideMeters);
+  auto const selectCandidates = [&rect, &candidates](FeatureType & ft) {
+    ft.ParseGeometry(FeatureType::BEST_GEOMETRY);
+    ft.ForEachPoint(
+        [&rect, &candidates](m2::PointD const & candidate) {
+          if (rect.IsPointInside(candidate))
+            candidates.emplace_back(candidate);
+        },
+        scales::GetUpperScale());
+  };
+
+  m_index.ForEachInRect(selectCandidates, rect, scales::GetUpperScale());
+
+  // TODO: Move this to a separate stage.
+  // 1030292476 Does not match. Some problem occur with poins.
+  // Either points duplicatate or something alike. Check thils
+  // later. The idea to fix this was to move SortUnique to the stage
+  // aftex enriching with projections.
+
+  my::SortUnique(candidates,
+                 [&p](m2::PointD const & a, m2::PointD const & b) {
+                   return MercatorBounds::DistanceOnEarth(a, p) <
+                          MercatorBounds::DistanceOnEarth(b, p);
+                 },
+                 [](m2::PointD const a, m2::PointD const & b) { return a == b; });
+
+  LOG(LDEBUG, (candidates.size(), "candidate points are found for point", p));
```

Comment:
What's the reason to print p as mercator there and as lat-lon later?

Reply:
This is for debugging. I'll remove all of it in the final version. The reason behind using LatLon
is that it gives a way to find the point on some map and have an idea of what is going on.

Reply:
So, why not to print p always as lat-lon?